server/controllers/personal/personalController.ts

```diff
@@ -239,32 +284,52 @@ export default class PersonalController {
             errors: hasLength(errors) ? errors : [],
             fieldName: 'weight',
             fieldValue: fieldValueFlash.length > 0 ? fieldValueFlash[0] : prisonPerson?.physicalAttributes.weight,
+            miniBannerData: miniBannerData(prisonerData),
           })
         },
 
         submit: async (req: Request, res: Response, next: NextFunction) => {
           const { prisonerNumber } = req.params
           const { clientToken } = req.middleware
-          const { editField } = req.body
+          const { kilograms } = req.body
           const prisonPerson = await this.personalPageService.getPrisonPerson(clientToken, prisonerNumber, true)
 
-          const weight = parseInt(editField, 10)
-          if (Number.isNaN(weight) || weight <= 0) {
-            req.flash('fieldValue', editField)
-            req.flash('errors', [{ text: 'Enter a number greater than 0' }])
+          const weight = parseInt(kilograms, 10)
+          const validatedInput = (): { valid: boolean; errorMessage?: string } => {
```

Comment:
It might be worth moving the validation out of the controller and look to see if we can make use of the `validationMiddleware` that we use for case notes already?

Reply:
If the validationMiddleware is adapted a bit, we could get it to add to `req.flash` too